src/components/SearchReps.vue

```diff
@@ -188,18 +175,27 @@ export default {
             this.step2 = step2
         },
         CheckInputContent: function () {
-          if (this.postalCode !== '') {
-              this.hasContent = true
-          } else {
-              this.hasContent = false
-          }
+
+            if (this.searchText !== '') {
+                this.hasContent = true
+            } else {
+                this.hasContent = false
+            }
         },
         async CreateRepList() {
             try {
                 this.$store.commit('setGenericValue', { key: 'zipcode', value: this.postalCode })
-                const res = await axios.get(
-                    '/api/representatives/' + this.postalCode
+                // check postal code is valid with regex
+                let res = ''
+                let isPostalCodeValid =  /(^\d{5}$)|(^\d{5}-\d{4}$)/.test(this.searchText);
+                let streetAddressValid = /^(\d{3,})\s?(\w{0,5})\s([a-zA-Z]{2,30})\s([a-zA-Z]{2,15})\.?\s?(\w{0,5})$/.test(this.searchText);
+                if(isPostalCodeValid || streetAddressValid) {
+                  console.log('valid input to get representatives')
+                  res = await axios.get(
+                    '/api/representatives/' + this.searchText
```

Comment:
I'm not sure if if this will work as written.  The search text should not be passed as part of the route, it should be a request parameter.  It was a bad pattern to use with ZIP codes as well, but it worked because none of the characters in a ZIP code need to be encoded in a URL.  Addresses can have spaces and other blank space in them, which needs to be encoded in a URL (e.g. a space becomes `%20`).

So you'll need to change the call to get, and change the route to take `searchText` as either a parameter in the body of the request or a URL query parameter (e.g. `&searchText=<address>`).  The former is a better option, as encoding just adds extra complexity and the address could be quite long.

Reply:
Hello, @thyeggman. I have updated the input text to be a URL query parameter when it is a street address (or the string is valid for a street address with the initial regex check), keeping in mind that in the near future, @paramsiddharth and @manishapriya94 seem to suggest we can have two input text fields and therefore keep two parameters. 


Second, @paramsiddharth and you seem to differ on whether the postal code should be a URL query parameter or not. If we change them to be part of the body, I need to make a POST request instead and add each parameter to the body of the request. It is important for me that you both come to a single conclusion, as I have already stated my position regarding it.